python/mp/effects/bounce.py

```diff
@@ -14,16 +14,15 @@ class Bounce(effect.Effect):
     # Wish there were a better way than requiring this every time
     dm = copy.deepcopy(effect.Effect.dm)
 
-    def __init__(self, bead_set, color=color.Color(), duration=None, direction=1):
-        super().__init__("bounce", bead_set, color=color, duration=duration)
+    def __init__(self, bead_set, color=color.Color(), direction=1, **kwargs):
+        super().__init__(name="bounce", bead_set=bead_set, color=color, **kwargs)
         self.direction = direction
         if (self.direction < 0):
             self.current = len(self.bead_list) - 1
         else:
             self.current = 0
 
-    def next(self, rosary):
-        super().next()
+    def next(self):
```

Comment:
No longer calling super().next() because in effect.py and rosary.py, you'll see that we're now calling `supernext()`, which in turn calls the specific `next()` method.